src/info/mod.rs

```diff
@@ -364,4 +371,52 @@ mod tests {
         );
         Ok(())
     }
+
+    type Result<T = ()> = std::result::Result<T, Box<dyn std::error::Error>>;
+
+    fn test_repo(name: &str) -> Result<Repository> {
+        let repo_path = git_testtools::scripted_fixture_repo_read_only(name)?;
+        let safe_repo = ThreadSafeRepository::open_opts(repo_path, open::Options::isolated())?;
+        Ok(safe_repo.to_thread_local())
+    }
+
+    #[test]
+    fn test_bare_repo() -> Result {
+        let repo = test_repo(&"bare_repo.sh")?;
+        match Info::init_repo_path(&repo) {
+            Ok(_info) => assert!(false, "oops, info was returned on a bare git repo"),
+            Err(error) => assert_eq!(
+                "please run onefetch inside of a non-bare git repository",
+                error.to_string()
+            ),
+        };
+        Ok(())
+    }
+
+    #[test]
+    fn test_language_repo() -> Result {
+        let repo_path = git_testtools::scripted_fixture_repo_read_only_with_args(
+            "language_repo.sh",
+            ["verilog"],
+        )
+        .unwrap();
+        let safe_repo =
+            ThreadSafeRepository::open_opts(repo_path.join("verilog"), open::Options::isolated())?;
+        let repo = safe_repo.to_thread_local();
+        let mut config = Config::parse_from(&["."]);
```

Comment:
Is this comment chain still relevant? https://github.com/o2sh/onefetch/pull/822#discussion_r994138849
Perhaps this should be an absolute path (defined at compile-time with macros). Or some refactoring to the implementation so that there's an easy way to create a `Config` struct besides parsing from command-line args.

Reply:
I only have one perspective and experience with this so am reluctant to do any refactoring without significant coaching ;-) It would be nice to create a new Config and then just create a new Info using the path returned from `git-testtools`.

@Byron would be able to speak about using absolute paths versus non absolute paths.

The rest of this is some review that I think helps:

This PR uses `git-testtools` approach of using scripts to create known repositories.
The directory `tests/fixtures/generated-do-not-edit` is a  new part of that method.

How to create and see a clean generated test directory:
```text
rm -fr tests/fixtures/generated-do-not-edit
cargo test -- --nocapture 
```

What does the generated directory look like?
`tests/fixtures/generated-do-not-edit` directory looks like this (only directories are shown, no files):
```text
tree -d tests/fixtures/generated-do-not-edit/
tests/fixtures/generated-do-not-edit/
├── bare_repo
│   └── 3723886778-unix
│       ├── hooks
│       ├── info
│       ├── objects
│       │   ├── info
│       │   └── pack
│       └── refs
│           ├── heads
│           └── tags
├── basic_repo
│   └── 2065824803-unix
└── language_repo
    └── 320037018-unix
        └── verilog
```

@spenserblack 
Maybe using the shell script to generate files is helpful because the generated path has some dependencies not related to an absolute path?

Here is a script created `verilog` test repo:

```text
tree tests/fixtures/generated-do-not-edit/language_repo
tests/fixtures/generated-do-not-edit/language_repo
└── 320037018-unix
    └── verilog
        ├── code.vg
        ├── expected_json
        └── expected_regex
```

The `language_repo` shell script was used on `unix` using a specific `320037018` script.

Here is what the tree looks like after adding another language, in this case I added `vhdl`. The test code for this likely belongs at the top level and might be useful to add a `expected_ansi_art` file to the scripts abilities.

```text
tree tests/fixtures/generated-do-not-edit/language_repo
tests/fixtures/generated-do-not-edit/language_repo
└── 320037018-unix
    ├── verilog
    │   ├── code.vg
    │   ├── expected_json
    │   └── expected_regex
    └── vhdl
        ├── code.vhdl
        ├── expected_json
        └── expected_regex
```

Reply:
> so am reluctant to do any refactoring without significant coaching ;-)

So this function currently only exists in a test mod: 
https://github.com/o2sh/onefetch/blob/3a3b1c6c04907d5461deb6ed609a2543f5be3776/src/cli.rs#L259-L287

But I think there is value in moving this logic to an implementation of the `Default` trait, making `Default::default()` available throughout the source code where a `Config` needs to be constructed (especially tests).

With `Default` implemented, you can set only the config values that you *need* to set, and leave everything else as a "do not care" default value.

Hopefully [this example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&code=%23!%5Ballow(dead_code)%5D%0A%23%5Bderive(Debug)%5D%0Aenum%20Color%20%7B%0A%20%20%20%20Black%2C%0A%20%20%20%20White%2C%0A%20%20%20%20RGB(u8%2C%20u8%2C%20u8)%2C%0A%20%20%20%20%2F%2F%20etc.%0A%7D%0A%0A%23%5Bderive(Debug)%5D%0Astruct%20Output%20%7B%0A%20%20%20%20foreground%3A%20Color%2C%0A%20%20%20%20background%3A%20Color%2C%0A%20%20%20%20text%3A%20String%2C%0A%7D%0A%0Aimpl%20Default%20for%20Output%20%7B%0A%20%20%20%20fn%20default()%20-%3E%20Output%20%7B%0A%20%20%20%20%20%20%20%20Output%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20foreground%3A%20Color%3A%3AWhite%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20background%3A%20Color%3A%3ABlack%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20text%3A%20Default%3A%3Adefault()%2C%0A%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%7D%0A%7D%0A%0Afn%20main()%20%7B%0A%20%20%20%20let%20output%20%3D%20Output%7Btext%3A%20%22Hello%2C%20World!%22.into()%2C%20..Default%3A%3Adefault()%7D%3B%0A%20%20%20%20println!(%22output%20struct%3A%20%7B%3A%3F%7D%22%2C%20output)%3B%0A%7D) shows how this can be used.

Reply:
Nice idea, removes the need for the `.` weirdness when constructing the Config object.
[2f04c83](https://github.com/o2sh/onefetch/pull/827/commits/2f04c8354fe046e3e436f33cd8e9debbce36e64e)